File: clojure/examples/fizzbuzz.clj
(defn fb [x]
  (if (== 0 (mod x 3))
    (if (== 0 (mod x 5))
      "FizzBuzz"
      "Fizz")
    (if (== 0 (mod x 5))
      "Buzz"
      x)))

(defn fizzbuzz []
  (map fb (range 1 100)))

File: haskell/examples/Fizzbuzz.hs
fizzbuzz :: [String]
fizzbuzz = map fb [1..100]

fb :: Int -> String
fb x | x `mod` 15 == 0 = "FizzBuzz"
fb x | x `mod` 5 == 0 = "Fizz"
fb x | x `mod` 3 == 0 = "Buzz"
fb x = show x